src/services/__tests__/cgnOperatorSearchService.test.ts

```diff
@@ -0,0 +1,276 @@
+import * as t from "io-ts";
+import { NonNegativeInteger } from "italia-ts-commons/lib/numbers";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import {
+  OfflineMerchantSearchRequest,
+  OrderingEnum
+} from "../../../generated/io-cgn-operator-search-api/OfflineMerchantSearchRequest";
+import { OnlineMerchantSearchRequest } from "../../../generated/io-cgn-operator-search-api/OnlineMerchantSearchRequest";
+import { ProductCategoryEnum } from "../../../generated/io-cgn-operator-search-api/ProductCategory";
+import { CgnOperatorSearchAPIClient } from "../../clients/cgn-operator-search";
+import CgnOperatorSearchService from "../cgnOperatorSearchService";
+
+const mockGetMerchant = jest.fn();
+const mockGetOfflineMerchants = jest.fn();
+const mockGetOnlineMerchants = jest.fn();
+
+mockGetMerchant.mockImplementation(() => t.success({ status: 200, value: {} }));
```

Comment:
maybe we can add a value and check it into "should handle a success response"

Reply:
these values are very mutant in this moment, we just check for the call to get a 200, decoding should just work fine since models are the same.

Reply:
The test should check that the type will remain the same, and if for some reason one of them change rise an error.